### ISSUE TITLE
Remove leading/trailing spaces on all metadata entry

### ISSUE THREAD

**Issue by jrochkind** [sciencehistory/scihist_digicoll#1346]
* in existing data
* on entry of new data

came out of discussion in #1320

**Comment by jrochkind** [comment]
Hey @apinkney0696 , are there particular fields that this has come up in, that you care about especially/only? (Subject and creator? others?) -- or is it best if we can make this happen for "every" field?

**Comment by apinkney0696** [comment]
Good question! The fields **Subject**, **Creator**, **Place**, and **Language** are most important to me because the extra spaces are creating new linked terms. If it's easier to do all the fields at once it shouldn't hurt anything, but those fields are the main concerns!

**Comment by jrochkind** [comment]
In looking at existing data in production, I'm actually seeing them in almost all fields. Here are the counts:

creator: 137
description: 928 (includes descirptions that just end in newlines, probably doens't really count)
external_id: 54
rights_holder: 23
place: 11
admin_note: 100 (includes notes that just end in newlines, probably doens't really count)
extent: 69
inscription: 40
inscription.location: 13
subject: 103
digitization_funder: 14
physical_container.shelfmark: 15,
physical_container.box: 9
language: 21, 
physical_container.page: 5
additional_title: 2, 
date_of_work.note: 6
series_arrangement: 1
provenance: 1

I see what you mean about it only really matters for the ones used for faceting.... it's maybe a bit more straightforward to do it for specific specified fields... but i think it may make sense just to do it for all fields that we can, since it seems to happen across them. 

I wonder where these all come from! Copy/paste into the field, you think mostly?

**Comment by apinkney0696** [comment]
Yeah, I think they must come from copy/paste or just typing too quickly I suppose. 

But wow, I had a hunch there were a lot in Subject and Creator!

I'm happy to get rid of them in all the fields to avoid any potential issues in the future. Evidently, it's a very common mistake.

**Comment by jrochkind** [comment]
The "multi-line" inputs -- description, provenance, and admin note -- can sometimes end in a newline, like you hit return at the end of the last line. 

While this isn't really a problem, the easiest way of approaching this would get rid of those too. Which I think would be fine? Alternately, we might end up with newlines at the end of Subject, Creator, etc -- where we really _don't_ want them if they sneak in somehow.  Or have to individually configure different for differnet fields, which is more error-prone. 

So I am leaning toward trimming trailing newlines even from the multi-line entries.

**Comment by apinkney0696** [comment]
I'm totally okay with trimming trailing newlines from multi-line entries. Shouldn't be a problem at all.

**Comment by jrochkind** [comment]
Okay, @apinkney0696 !

On **staging**, I ran a routine to eliminate leading/trailing whitespace on all fields *ignoring* `description` and `admin_notes`. 

That touched 473 records. 

I always get nervous bulk editing data like this; but it's not really realistic for us to "double check" all 473 records! 

But maybe we should spot check a few? I'm not sure. Can you help me on staging just take a look and confirm that it looks good to you?

Here's a list of all 473 records on staging touched, along with what fields were found to have whitespace needing removal: https://gist.github.com/jrochkind/4b0de8d267203d2eca9356f8657ebafb
